src/main/java/org/opensearch/security/dlic/rest/api/MigrateApiAction.java

```diff
@@ -101,34 +91,33 @@ public List<Route> routes() {
     }
 
     @Override
-    protected Endpoint getEndpoint() {
-        return Endpoint.MIGRATE;
+    protected CType getConfigType() {
+        return null;
     }
 
     @Override
-    protected boolean hasPermissionsToCreate(
-        final SecurityDynamicConfiguration<?> dynamicConfigFactory,
-        final Object content,
-        final String resourceName
-    ) {
-        return true;
+    protected void consumeParameters(final RestRequest request) {
+        // not needed
+    }
+
+    private void migrateApiRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
+        requestHandlersBuilder.allMethodsNotImplemented().override(Method.POST, (channel, request, client) -> migrate(channel, client));
     }
 
     @SuppressWarnings("unchecked")
-    @Override
-    protected void handlePost(RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+    protected void migrate(final RestChannel channel, final Client client) throws IOException {
 
-        final Version oldestNodeVersion = cs.state().getNodes().getMinNodeVersion();
+        final Version oldestNodeVersion = clusterService.state().getNodes().getMinNodeVersion();
```

Comment:
Do we need such check in the main branch?

Reply:
That might be a good one to forward port, nice catch

Reply:
@peternied I will move it to main